.ci/register_package.jl

```diff
@@ -48,26 +49,57 @@ function download_cached_binaries(download_dir, platforms)
     end
 end
 
+function download_binaries_from_release(download_dir, platforms)
+    probe_platform_engines!(;verbose=verbose)
+
+    # The version of the latest build is one build number less than the next one
+    # (pretty obvious, isn't it?).  Since we're supposed to be rebuilding the
+    # latest version, make sure the next build version is greater than 0
+    build_version.build[1] > 0 || error("The next version must have build number greater than 0")
+    latest_build_version = VersionNumber(build_version.major, build_version.minor, build_version.patch,
+                                         build_version.prerelease, build_version.build .- 1)
+    tag = "$(name)-v$(latest_build_version)"
+
+    for platform in platforms
+        filename = "$(name).v$(version).$(triplet(platform)).tar.gz"
+        url = "https://github.com/$(repo)/releases/download/$(tag)/$(filename)"
+        PlatformEngines.download(url, joinpath(download_dir, filename); verbose=verbose)
+    end
+end
+
 # Filter out build-time dependencies also here
 for json_obj in [merged, objs_unmerged...]
     json_obj["dependencies"] = Dependency[dep for dep in json_obj["dependencies"] if !isa(dep, BuildDependency)]
 end
+skip_build = get(ENV, "SKIP_BUILD", "false") == "true"
 mktempdir() do download_dir
     # Grab the binaries for our package
-    download_cached_binaries(download_dir, merged["platforms"])
+    if skip_build
+        # We only want to update the wrappers, so download the tarballs from the
+        # latest build.
+        download_binaries_from_release(download_dir, merged["platforms"])
+    else
+        # We are going to publish the new binaries we've just baked, take them
+        # out of the cache while they're hot.
+        download_cached_binaries(download_dir, merged["platforms"])
+    end
     
     # Push up the JLL package (pointing to as-of-yet missing tarballs)
     tag = "$(name)-v$(build_version)"
     upload_prefix = "https://github.com/$(repo)/releases/download/$(tag)"
 
+    # If we didn't rebuild the tarballs, save the original Artifacts.toml
+    artifacts_toml = skip_build ? read(joinpath(code_dir, "Artifacts.toml"), String) : ""
     # This loop over the unmerged objects necessary in the event that we have multiple packages being built by a single build_tarballs.jl
     for (i,json_obj) in enumerate(objs_unmerged)
         from_scratch = (i == 1)
         BinaryBuilder.rebuild_jll_package(json_obj; download_dir=download_dir, upload_prefix=upload_prefix, verbose=verbose, lazy_artifacts=json_obj["lazy_artifacts"], from_scratch=from_scratch)
     end
+    # If we skipped the build, restore Artifacts.toml
+    skip_build && write(joinpath(code_dir, "Artifacts.toml"), artifacts_toml)
```

Comment:
TBH, I would find this easier to understand if it was in an `if/else`.  E.g.:

```
if skip_build
    write(...)
else
    upload_to_github_releases()
end
```

Reply:
When I did this yesterday it was too late to notice such an obvious improvement :slightly_smiling_face:   Thanks for pointing it out!